typescript/test/unit/baseline.ts

```diff
@@ -79,10 +78,57 @@ describe('CodeGeneratorTest', () => {
           `-I${PROTOS_DIR} ` +
           ECHO_PROTO_FILE
       );
-      assert.strictEqual(
-        fs.readFileSync(GENERATED_CLIENT_FILE).toString(),
-        fs.readFileSync(CLIENT_LIBRARY_BASELINE).toString()
-      );
+      const outputFiles = fs.readdirSync(OUTPUT_DIR);
+      // store every item (file or directory with full path in output dir and baseline dir) in stack, pop once a time
+      const protoItemStack: Item[] = [];
+      outputFiles.forEach(file => {
+        const fileFullPath = path.join(OUTPUT_DIR, file);
+        const baselinePath = path.join(BASELINE_DIR, file);
+        protoItemStack.push(new Item(file, fileFullPath, baselinePath));
+      });
+      console.warn(protoItemStack);
+      while (protoItemStack.length !== 0) {
+        const item = protoItemStack.pop();
+        if (!item) continue;
```

Comment:
how is this condition possible?

Reply:
I think the type checking complains here that `item` might be undefined, so I added this condition to make sure it's passing the check.